nums/numpy/api.py

```diff
@@ -258,6 +258,50 @@ def logspace(start, stop, num=50, endpoint=True, base=10.0, dtype=None, axis=0):
         ba = ba.astype(dtype)
     return ba
 
+############################################
+# Matrix Ops
+############################################
+
+def tensordot(x1: BlockArray, x2: BlockArray,
+        out: BlockArray = None, where=True, **kwargs) -> BlockArray:
+    if out is not None:
+        raise NotImplementedError("'out' argument is not yet supported for tensordot API.")
+    axes = kwargs.pop('axes', 2)
+    return _instance().tensordot(arr_1=x1,
+                                 arr_2=x2,
+                                 axes=axes)
+
+def matmul(x1: BlockArray, x2: BlockArray,
+        out: BlockArray = None, where=True, **kwargs) -> BlockArray:
+    if out is not None:
+        raise NotImplementedError("'out' argument is not yet supported for tensordot API.")
+    return _instance().matmul(arr_1=x1,
+                              arr_2=x2)
+
+############################################
+# Shape Ops
+############################################
+
+def ndim(x: BlockArray):
+    return x.ndim
+
+def metadata(x: BlockArray):
```

Comment:
Is this part of the NumPy API?

Reply:
I think it is part of autograd's numpy api. maybe it looks better to add this feature in autograd.